plugins/modules/packaging/os/copr.py

```diff
@@ -0,0 +1,496 @@
+            state["state"] = bool(disabled)
+        elif self.state == "absent":
+            removed = self._remove_repo()
+            state["msg"] = "absent"
+            state["state"] = bool(removed)
+        return state
+
+    @staticmethod
+    def _compare_repo_content(repo_filename_path, repo_content_api):
+        """Compare the contents of the stored repository with the information from the server.
+
+        Args:
+            repo_filename_path: Path to repository.
+            repo_content_api: The information about the repository from the server.
+
+        Returns:
+            True, if the information matches, False otherwise.
+        """
+        if not os.path.isfile(repo_filename_path):
+            return False
+        with open(repo_filename_path, "r") as file:
+            repo_content_file = file.read()
+        return repo_content_file == repo_content_api
+
+    @staticmethod
+    def chroot_conf():
+        """Obtain information about the distribution, version, and architecture of the target.
+
+        Returns:
+            Chroot info in the form of distribution-version-architecture.
+        """
+        (distribution, version, codename) = distro.linux_distribution(full_distribution_name=False)
+        base = CoprModule.get_base()
+        return "{0}-{1}-{2}".format(distribution, version, base.conf.arch)
+
+    @staticmethod
+    def _sanitize_username(user):
+        """Modify the group name.
+
+        Args:
+            user: User name.
+
+        Returns:
+            Modified user name if it is a group name with @.
+        """
+        if user[0] == "@":
+            return "group_{0}".format(user[1:])
+        return user
+
+
+def run_module():
+    """The function takes care of the functioning of the whole ansible copr module."""
+    module_args = dict(
+        host=dict(type="str", default="copr.fedorainfracloud.org"),
+        protocol=dict(type="str", default="https"),
+        name=dict(type="str", required=True),
+        state=dict(type="str", choices=["enabled", "disabled", "absent"], default="enabled"),
+        chroot=dict(type="str"),
+    )
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+    params = module.params
+
+    if not HAS_DNF_PACKAGES:
+        module.fail_json(msg=missing_required_lib("dnf"), exception=DNF_IMP_ERR)
+
+    CoprModule.ansible_module = module
+    copr_module = CoprModule(
+        host=params["host"],
+        name=params["name"],
+        state=params["state"],
+        protocol=params["protocol"],
+        chroot=params["chroot"],
+        check_mode=module.check_mode,
+    )
+    state = copr_module.run()
+
+    info = "Please note that this repository is not part of the main distribution"
+
+    if module.check_mode:
+        module.exit_json(
+            state=params["state"],
+            chroot=params["chroot"],
+            host=params["host"],
+            changed=state["state"],
+        )
```

Comment:
Why does check mode reutnr `state`, `chroot` and `host`, while the other cases don't?